src/ripple/rpc/impl/RPCHelpers.cpp

```diff
@@ -401,10 +401,17 @@ addPaymentDeliveredAmount(Json::Value& meta, RPC::Context& context,
     if (! transaction)
         return;
 
-    auto serializedTx = transaction->getSTransaction ();
-    if (! serializedTx || serializedTx->getTxnType () != ttPAYMENT)
+    auto const serializedTx = transaction->getSTransaction ();
+    if (! serializedTx)
         return;
 
+    {
+        // Only include this field for Payment and CheckCash transactions.
```

Comment:
Is it a bad thing to add delivered amount to Escrow or Payment Channels? I should hope that _all_ transactions which can deliver currency should have it, just for simplicity in processing transaction metadata. (I mean, that would be outside the scope of the Checks amendment to change that here, I guess...)

Reply:
Excellent question, @mDuo13.  This specific topic came up when I was talking with Dave Cohen in San Francisco last December.  Dave's position was that all transactions that transfer funds (including all `Payment` transactions, not just the partials) should put their final outcome in a `delivered_amount` field.  This would have the potential to vastly simplify the computational hoops that an exchange has to jump through.

If we can simplify the life of an exchange I'm all for it.  However we need to think hard about some transactions like `EscrowCreate`.  From an _exchange's_ perspective does that XRP still belong to the owning account, even though the XRP is, for now, locked away in the `Escrow`?  I think we'd want to be _really_ clear about the information we intend to convey before we make a change of this nature.

FWIW, since the `delivered_amount` is synthesized, we could make this change at any time without an Amendment.  I just want to make sure we've thought it through really carefully (and consulted an expert on exchanges) before we make the change.

Reply:
That's why it's worth thinking about.  An `EscrowCreate` removes available XRP from an account.  Perhaps that account can no longer pay fees after the `EscrowCreate`?  I just want to make sure that any features we add align with the needs to the customers who we hope to have using those features.